### PR TITLE
Feature - PhongHuynh - SwiftBasic

### PR DESCRIPTION
**Pull request by mvn-phonghuynh-dn**

### CODE REVIEW

File: 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,162 @@
+import UIKit
+
+// Bai tap 1
+func dienTICHVATHETICHHINHCAU(r: Float) -> (s: Float, v: Float) {
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift:4]: Start of function with 1 verb. 
**Please check it in all your code.**

File: 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    }
+}
+hePhuongTrinh(a1: 1, a2: 1, b1: 1, b2: 2, c1: 2, c2: 2)
+
+//Bai tap 4
+func fibonacci(n: Int) -> Int {
+  if n <= 2 {
+    return 1
+  }
+    return fibonacci(n: n - 1) + fibonacci(n:n - 2)
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift:60]: Please check format code (wrong tab size)

File: 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+}
+print("tong cua n so fibonacci dau tien \(tinhTong(n: 3))")
+
+// Bai tap 4
+func nguyenTo(n: Int) -> Int {
+    var temp: Int = Int(sqrt(Double(n)))
+    for i in 2...temp {
+        if n % i == 0{
+        return 0
+        }
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift:79]: Check format code

File: 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    }
+    return 1
+}
+
+func kiemTra(n: Int) -> Int {
+    var temp = [Int]()
+    var tong1: Int = 0
+    var tong2: Int = 0
+    for i in 0...n/2 {
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift:88]: Add space before and after of "/"

File: 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    for i in 2...temp {
+        if n % i == 0{
+        return 0
+        }
+    }
+    return 1
+}
+
+func kiemTra(n: Int) -> Int {
+    var temp = [Int]()
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift:85]: Name of array must be have suffix are "s/es/array,..."

File: 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    var tong1: Int = 0
+    var tong2: Int = 0
+    for i in 0...n/2 {
+        tong1+=temp[i] - 0
+    for i in 0...n {
+        tong2+=temp[i]-0
+    return tong1
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift:92]: Check format code

File: 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+    }
+    return tong2
+
+}
+
+func lietKe(n: Int) -> Int {
+    for i in 100001..<999999 {
+        }
+    return n
+    }
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift:103]: Check format code

File: 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+//Bai tap 5
+func timKiemChuoi(arraycon: String, arrayme: String) {
+    var string = "abaaaabbbabababbaaabababa"
+    var dem = 0
+    for value in string {
+        if string.contains("ab"){
+            dem+=1
+            print("so lan xuat hien chuoi con la \(dem)")
+        }
+    }
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift:115]: Wrong code, please use parameter

File: 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+  var listArray: [Int] = array
+  var shuffleArray: [Int] = []
+  while listArray.count > 0 {
+    if let randomElement = listArray.randomElement(), let index = listArray.firstIndex(of: randomElement) {
+      shuffleArray.append(randomElement)
+      listArray.remove(at: index)
+    }
+  }
+  return shuffleArray
+}
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift:135]: Check format code (space)

File: 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+    let listArray: [Int] = array
+    return listArray.shuffled()
+}
+print(hoanViRanDom(array: [0, 1, 2, 3, 4 , 5, 6, 7, 8, 9]))
+
+//Bai tap 7
+func thayThe(inputArray: [Int], elemToReplace: Int, substitutionElem: Int) -> [Int] {
+    var newArray = inputArray
+    for i in 0..<inputArray.count {
+        if (inputArray[i] == elemToReplace){
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasic.playground/Contents.swift:149]: No need "(" and ")", add space before "{"

**Review by mvn-bienle-dn — COMMENTED** [review]
@mvn-phonghuynh-dn Please check my comments.
Thank you!

File: 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift [hunk 11]
```diff
@@ -0,0 +1,130 @@
+import UIKit
+
+// Bai tap 1
+func tinhHINHCAU(r: Float) -> (s: Float, v: Float) {
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift:4]: Rename with camel format: tinhHINHCAU ~ tinhHinhCau
**Please check it in all your code.**

File: 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+    let pi = 3.14
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift:5]: No need to define, use Double.pi or Float.pi

File: 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+}
+giaiPTBAC2(a: 1, b: -6, c: 5)
+
+//Bai tap 3
+func hePhuongTrinh(a1: Int, a2: Int, b1: Int, b2: Int, c1: Int, c2: Int) {
+    let D = a1 * b2 - a2 * b1
+    let Dx = c1 * b2 - c2 * b1
+    let Dy = a1 * c2 - a2 * c1
+    if D == 0 {
+        if Dx == Dy {
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift:43]: Name of variable have prefix with lowercase letter.
Ex: d, dx, dX, dy, dY.
**Please check it in all your codes.**

File: 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+    }
+}
+hePhuongTrinh(a1: 1, a2: 1, b1: 1, b2: 2, c1: 2, c2: 2)
+
+//Bai tap 4
+func fibonacci(n: Int) -> Int {
+  if n <= 2 {
+    return 1
+  }
+  return fibonacci(n: n - 1) + fibonacci(n:n - 2)
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift:61]: Wrong logic, please research what is Fibonacci and follow required.

File: 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+func nguyenTo(n: Int) -> Int {
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift:64]: Please define name function begin by verb.
Ex: timNguyenTo/timSoNguyenTo...

File: 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+    let temp: Int = Int(sqrt(Double(n)))
+    
+    for i in 2...temp {
+        if n % i == 0{
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift:68]: Add space before "{"

File: 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift [hunk 17]
```diff
@@ -0,0 +1,130 @@
...
+        return 0
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift:69]: Add a tab space before return. (Check format code)

File: 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+        }
+    }
+    return 1
+}
+
+func tinhTong(n: Int) -> Int {
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift:75]: Rename, exactly what the function do.
Ex: tinhTongFibonacci,...

File: 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift [hunk 19]
```diff
@@ -0,0 +1,130 @@
...
+    var s: Int = 0
+    
+    for i in 1...n {
+        print("so fibonacci thu \(i) la \(fibonacci(n: i))")
+        s+=fibonacci(n: i)
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift:80]: Check format code

File: 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift [hunk 20]
```diff
@@ -0,0 +1,130 @@
...
+    }
+    return s
+}
+print("tong cua n so fibonacci dau tien \(tinhTong(n: 3))")
+
+//Bai5 tim so luong chuoi con trong chuoi me
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift:86]: Add space after "//"

File: 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift [hunk 21]
```diff
@@ -0,0 +1,130 @@
...
+func countSubString(_ me: String, _ con: String) -> Int {
+  return me.components(separatedBy: con).count - 1
+}
+print("so luong chuoi con la \(countSubString("ababababababaab", "ab"))")
+
+// Bai 6: hoan vi
+ func hoanViRanDom2(array: [Int]) -> [Int] {
```
Comment (mvn-bienle-dn) [inline 02_SwiftBasic/PhongHuynh/SwiftBasicPlayground.playground/Contents.swift:93]: Remove space before "func"

**Review by mvn-bienle-dn — COMMENTED** [review]
@mvn-phonghuynh-dn Please check my comments.
Thank you!

**Comment by mvn-quedinh-dn** [comment]
LGTM!

**Pull request merged**